src/slider/slider-labels.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { useMobile } from '../internal/hooks/use-mobile/index.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export interface SliderLabelProps {
+  min: number;
+  max: number;
+  referenceValues?: ReadonlyArray<number>;
+  valueFormatter?: (value: number) => string;
+  labelsId: string;
+}
+export default function SliderLabels({ min, max, referenceValues, valueFormatter, labelsId }: SliderLabelProps) {
+  const isMobile = useMobile();
+  const LABEL_THRESHOLD = isMobile ? 5 : 10;
```

Comment:
Does LABEL_THRESHOLD represent the max number of labels to show? If so, would it make sense to rename the variable to e.g. `VISIBLE_LABELS_COUNT`?